### PR TITLE
Some build fu

### PR DESCRIPTION
**Pull request by matzew**
this works, for GCM file, with bogus content, @secondsun @danielpassos

### CODE REVIEW

**Comment by matzew** [comment]
Failures are those that we already have on different env.:

```
The command "echo no | android create avd --force -n test -t "$ANDROID_EMULATOR" --abi "$ANDROID_ABI"" failed and exited with 1 during .
```

File: .travis.yml [hunk 1]
```diff
@@ -45,6 +45,7 @@ install:
   - ./gradlew clean
 
 before_script:
+  - cp travis-google-services.json app/google-services.json
```
Comment (secondsun) [inline .travis.yml]: -1.  Instead of copying the file we should be using a build flavor.  PR incoming

File: .travis.yml [hunk 2]
```diff
@@ -49,7 +49,7 @@ before_script:
   - adb shell input keyevent 82 &
 
 script:
-  - ./gradlew build connectedCheck
+  - ./gradlew connectedTravisDebugAndroidTest
```
Comment (danielpassos) [inline .travis.yml:52]: Hey @secondsun i'm curious what does it do

File: app/build.gradle [hunk 3]
```diff
@@ -20,7 +20,11 @@ android {
         abortOnError false
     }
 
-    buildTypes {
+    productFlavors {
+        travis{}
```
Comment (danielpassos) [inline app/build.gradle:24]: Hey @secondsun i'm curious what does it do

Comment (secondsun) [inline app/build.gradle:24]: In this case it creates a product flavor, travis.  Product flavors are basically maven build profiles but for Android.  In this case it lets us supply a different google-services.json file to use when we are running the travis build.

File: .travis.yml (hunk 2 above)
Comment (secondsun) [inline .travis.yml:52]: Product flavors take advantage of Gradle's slippery nature.  

So connectedCheck runs the default tests on an emulator.  To use the Travis build flavor and run the Travis tests I use connectedTravisDebugAndroidTest.  It is a bunch of properties dynamically chained together in the task name.

**Pull request merged**